src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+
+import * as _ from 'lodash';
+import { ToastModule } from 'ng2-toastr';
+import { BsModalRef, ModalModule } from 'ngx-bootstrap';
+
+import { NotificationType } from '../../../../shared/enum/notification-type.enum';
+import { NotificationService } from '../../../../shared/services/notification.service';
+import { SharedModule } from '../../../../shared/shared.module';
+import { configureTestBed } from '../../../../shared/unit-test-helper';
+import { OsdFlagsModalComponent } from './osd-flags-modal.component';
+
+function getFlagsArray(component: OsdFlagsModalComponent) {
+  const allFlags = _.cloneDeep(component.allFlags);
+  allFlags['purged_snapdirs'].value = true;
+  allFlags['pause'].value = true;
+  return _.toArray(allFlags);
+}
+
+describe('OsdFlagsModalComponent', () => {
+  let component: OsdFlagsModalComponent;
+  let fixture: ComponentFixture<OsdFlagsModalComponent>;
+  let httpTesting: HttpTestingController;
+
+  configureTestBed({
+    imports: [
+      ReactiveFormsModule,
+      ModalModule.forRoot(),
+      SharedModule,
+      HttpClientTestingModule,
+      ToastModule.forRoot()
+    ],
+    declarations: [OsdFlagsModalComponent],
+    providers: [BsModalRef]
+  });
+
+  beforeEach(() => {
+    httpTesting = TestBed.get(HttpTestingController);
+    fixture = TestBed.createComponent(OsdFlagsModalComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should finish running ngOnInit', () => {
+    fixture.detectChanges();
+
+    const flags = getFlagsArray(component);
+
+    const req = httpTesting.expectOne('api/osd/flags');
+    req.flush(['purged_snapdirs', 'pause', 'foo']);
+
+    expect(component.flags).toEqual(flags);
+    expect(component.unknownFlags).toEqual(['foo']);
+  });
+
+  it('should run submitAction', () => {
+    let notificationType: NotificationType;
+    const notificationService = TestBed.get(NotificationService);
+    spyOn(notificationService, 'show').and.callFake((type) => {
+      notificationType = type;
+    });
+
+    const bsModalRef = TestBed.get(BsModalRef);
+    spyOn(bsModalRef, 'hide').and.callThrough();
+
+    component.flags = getFlagsArray(component);
+    component.unknownFlags = ['foo'];
+
```

Comment:
Both submit tests share a equal code base as they both test the submit behavior. I would suggest putting them into their own `describe`. I just tested it and it works :) see here:

```
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.spec.ts
index bb8e907..13a6660 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-flags-modal/osd-flags-modal.component.spec.ts
@@ -58,43 +58,41 @@ describe('OsdFlagsModalComponent', () => {
     expect(component.unknownFlags).toEqual(['foo']);
   });
 
-  it('should run submitAction', () => {
+  describe('test submit', function() {
     let notificationType: NotificationType;
-    const notificationService = TestBed.get(NotificationService);
-    spyOn(notificationService, 'show').and.callFake((type) => {
-      notificationType = type;
+    let notificationService: NotificationService;
+    let bsModalRef: BsModalRef;
+
+    beforeEach(() => {
+      notificationService = TestBed.get(NotificationService);
+      spyOn(notificationService, 'show').and.callFake((type) => {
+        notificationType = type;
+      });
+
+      bsModalRef = TestBed.get(BsModalRef);
+      spyOn(bsModalRef, 'hide').and.callThrough();
+      component.unknownFlags = ['foo'];
     });
 
-    const bsModalRef = TestBed.get(BsModalRef);
-    spyOn(bsModalRef, 'hide').and.callThrough();
+    it('should run submitAction', () => {
+      component.flags = getFlagsArray(component);
+      component.submitAction();
+      const req = httpTesting.expectOne('api/osd/flags');
+      req.flush(['purged_snapdirs', 'pause', 'foo']);
+      expect(req.request.body).toEqual({ flags: ['pause', 'purged_snapdirs', 'foo'] });
 
-    component.flags = getFlagsArray(component);
-    component.unknownFlags = ['foo'];
-
-    component.submitAction();
-    const req = httpTesting.expectOne('api/osd/flags');
-    req.flush(['purged_snapdirs', 'pause', 'foo']);
-    expect(req.request.body).toEqual({ flags: ['pause', 'purged_snapdirs', 'foo'] });
-
-    expect(notificationType).toBe(NotificationType.success);
-    expect(component.bsModalRef.hide).toHaveBeenCalledTimes(1);
-  });
-
-  it('should hide modal if request fails', () => {
-    const notificationService = TestBed.get(NotificationService);
-    spyOn(notificationService, 'show').and.returnValue(true);
-
-    const bsModalRef = TestBed.get(BsModalRef);
-    spyOn(bsModalRef, 'hide').and.callThrough();
+      expect(notificationType).toBe(NotificationType.success);
+      expect(component.bsModalRef.hide).toHaveBeenCalledTimes(1);
+    });
 
-    component.flags = [];
-    component.unknownFlags = ['foo'];
+    it('should hide modal if request fails', () => {
+      component.flags = [];
+      component.submitAction();
+      const req = httpTesting.expectOne('api/osd/flags');
+      req.flush([], { status: 500, statusText: 'failure' });
 
-    component.submitAction();
-    const req = httpTesting.expectOne('api/osd/flags');
-    req.flush([], { status: 500, statusText: 'failure' });
-
-    expect(notificationService.show).toHaveBeenCalledTimes(0);
-    expect(component.bsModalRef.hide).toHaveBeenCalledTimes(1);
+      expect(notificationService.show).toHaveBeenCalledTimes(0);
+      expect(component.bsModalRef.hide).toHaveBeenCalledTimes(1);
+    });
   });
 });
```

Reply:
Done.